### PR TITLE
(10.0.x) ByteBufferPool and RetainableByteBufferPool now reset the buffer's endianness on release

### LINKED ISSUE #7243: Reset pooled ByteBuffer endianness
**Target Jetty version(s)**
9.4.x, 10.0.x, 11.0.x

**Enhancement Description**
User code may be passed `ByteBuffer`s that are pooled by Jetty.
They may use the buffer and change the endianness and then forget to reset it back, so the next usage of the buffer by Jetty may use the wrong endianness.

We should reset the endianness when we put back the buffer to the pool, like we already clear it, and like we already reset the interrupted status of pooled threads in the thread pool.

### PR DESCRIPTION
**Pull request by lorban**
See #7243.

### CODE REVIEW

File: jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java [hunk 1]
```diff
@@ -166,6 +167,23 @@ public static ByteBuffer copy(ByteBuffer buffer)
         return clone;
     }
 
+    /**
+     * Reset the buffer's endianness to {@link ByteOrder#BIG_ENDIAN}
+     * and clear the buffer to be empty in flush mode.
```
Comment (sbordet) [inline jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java]: Use 3rd person as per javadoc standards ("Resets", "clears"). Wrap with `<p>` the paragraphs.

File: jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java [hunk 2]
```diff
@@ -166,6 +167,23 @@ public static ByteBuffer copy(ByteBuffer buffer)
...
+     * The position and limit are set to 0;
```
Comment (sbordet) [inline jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java]: Why semicolon at end of sentence? Something was forgot or typo?

File: jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java [hunk 3]
```diff
@@ -166,6 +167,23 @@ public static ByteBuffer copy(ByteBuffer buffer)
...
+     *
+     * @param buffer The buffer to clear.
```
Comment (sbordet) [inline jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java]:
```suggestion
     * @param buffer the buffer to reset
```

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java (hunk 2 above)
Comment (lorban) [reply jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java]: This is an exact copy of `clear()`'s javadoc, with the extra _Reset the buffer's endianness..._ bits added on top. I'll fix both.

File: jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java (hunk 3 above)
Comment (lorban) [reply jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java]: Ok.

File: jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java (hunk 1 above)
Comment (lorban) [reply jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java]: Ok.

**Review by sbordet — APPROVED** [review]

**Pull request merged**